Fixes for story/rules reading and writing for both formats (#7033)
RasaHQ/rasa: changelog/7033.bugfix.md

> - Fix `YAMLStoryReader` not being able to represent OR statements in conversion mode.

Reviewer: ```suggestion
- Fix `YAMLStoryReader` not being able to represent `OR `statements in conversion mode.
```

Reviewer: let's also link to these doc page https://rasa.com/docs/rasa/stories#or-statements

> - Fix `YAMLStoryReader` not being able to represent [`OR` statements](https://rasa.com/docs/rasa/stories#or-statements) in conversion mode.

Reviewer: you can actually use relativ links here
```suggestion
- Fix `YAMLStoryReader` not being able to represent [`OR` statements](stories.mdx#or-statements) in conversion mode.
```

> - Fix `MarkdownStoryWriter` not being able to write stories with OR statements (when loaded in conversion mode).

Reviewer: ```suggestion
- Fix `MarkdownStoryWriter` not being able to write stories with `OR` statements (when loaded in conversion mode).
```
